Update documentation

conradolandia opened the pull request:
This PR improves QDarkStyleSheet documentation to reflect recent API changes and address missing user guidance.

## Changes made:

**Enhanced Main API Documentation** (`qdarkstyle/__init__.py`):
- Added comprehensive documentation for the `palette` parameter in `load_stylesheet()`
- Fixed contradictory Python version support information
- Added practical usage examples with DarkPalette/LightPalette
- Marked `pyside` parameter as deprecated

**Complete Palette System Documentation** (`docs/color_reference.rst`):
- Documented all palette constants (Background, Text, Accent colors)
- Added hexadecimal color values for Dark and Light themes
- Included usage examples in Python and SCSS/CSS

**Custom Palettes Creation Guide** (`docs/custom_palettes.rst`):
- Added guide for creating custom themes
- Complete workflow from creation to deployment
- Best practices and troubleshooting documentation

**Utils CLI Documentation** (`docs/scripts/qdarkstyle_utils.rst`):
- Documented `python -m qdarkstyle.utils` command-line interface
- Added practical examples for resource generation

## Technical improvements:
- Fixed Sphinx build warnings
- Added cross-references between sections
- Integrated new documentation into main index structure

## Impact:
- Addresses missing API documentation gaps
- Provides complete guidance for custom theme creation
- Improves developer experience with clear examples

conradolandia committed:
Fix some formating issues in documentation. Remove unused import in `run_ui_css_edition.py`

conradolandia committed:
Fix some formating issues in documentation. Remove unused import in `run_ui_css_edition.py`

conradolandia committed:
Update documentation to reflect API changes

conradolandia commented:
@dalthviz would you take a look at these updates to see if I'm missing something or got anything wrong?

conradolandia committed:
Correct most recent QDarkStyle version accepted to work with Python 2

dalthviz commented on docs/color_reference.rst:
```diff
@@ -65,6 +65,129 @@ B140        ``#F5FBFF`` |image31|
+=================== ======================= ======================= =========
+OPACITY_TOOLTIP     230                     230                     Tooltip opacity
+SIZE_BORDER_RADIUS  4px                     4px                     Border radius
+=================== ======================= ======================= =========
```
Seems like the `Value` column of this table is duplicated. Should this be something like:

```suggestion
=================== ======================= =========
Constant            Value                   Usage
=================== ======================= =========
OPACITY_TOOLTIP     230                     Tooltip opacity
SIZE_BORDER_RADIUS  4px                     Border radius
=================== ======================= =========
```

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -0,0 +1,503 @@
+    stylesheet = qdarkstyle.load_stylesheet(palette=MyCustomPalette)
+    
+    # Apply to your application
+    app.setStyleSheet(stylesheet)
```
Not completly sure if `load_stylesheet` actually works with custom palettes 🤔 From a quick check seems like that only supports the `DarkPalette` and `LightPalette` palette classes:

https://github.com/ColinDuquesnoy/QDarkStyleSheet/blob/6f3962e63749893f951be73ebfec93e83fffebf2/qdarkstyle/__init__.py#L294-L368

https://github.com/ColinDuquesnoy/QDarkStyleSheet/blob/6f3962e63749893f951be73ebfec93e83fffebf2/qdarkstyle/__init__.py#L205-L252

dalthviz reviewed (commented):
Thanks @conradolandia ! Left some initial suggestions/comments (a suggestion for a table with a duplicated column) and a comment on the documentation regarding the usage of `load_stylesheet` (I think that function doesn't support custom palettes).

conradolandia committed:
Fix errors in the documentation for custom palettes

conradolandia committed:
Correct some functionality examples

conradolandia committed:
Apply suggestions from code review

Co-authored-by: Daniel Althviz Moré <16781833+dalthviz@users.noreply.github.com>

conradolandia committed:
Rewrite custom palettes documentation and improve examples after first review

conradolandia commented:
Thank you for your suggestions, @dalthviz! I fully rewrote `docs/custom_palettes.rst` and tried to verify all functionality for the examples, please let me know if you think something is inaccurate and/or missing. Also modified the documentation in `load_stylesheet` to clarify the supported palettes. I don't understand very well this part, because we have `palette = kwargs.get('palette', None)` at line 387 and `elif 'palette' in kwargs or issubclass(arg, Palette)` at line 423, so I guess it *could*  take personalized palettes, or I'm wrong?

conradolandia committed:
Add missing element to dictionary in `Palette.to_dict()` method

dalthviz commented:
> Also modified the documentation in load_stylesheet to clarify the supported palettes. I don't understand very well this part, because we have palette = kwargs.get('palette', None) at line 387 and elif 'palette' in kwargs or issubclass(arg, Palette) at line 423, so I guess it could take personalized palettes, or I'm wrong?

From my check, although the public function `load_stylesheet` doesn't have logic to validate that an arbitrary palette definition is being passed, the private function `_load_styleheet` (that actually does the resources load) has this validation:

https://github.com/ColinDuquesnoy/QDarkStyleSheet/blob/6f3962e63749893f951be73ebfec93e83fffebf2/qdarkstyle/__init__.py#L240-L252

In that function, if you pass as `palette` value `None` the `DarkPalette` will be loaded by default and, in case you try to load passing as palette value a palette class definition with something different than `light` or `dark` as `ID` (i.e `DarkPalette` or `LightPalette` related `ID`s), a message will be printed indicating that the palette is not recognized and the Python process will be exited.

So basically `load_stylesheet` ends up only supporting for the `palette` kwarg the `None`, `DarkPalette` and `LightPalette` values (or at least palette class definitions that use those palettes IDs) and can only load `DarkPalette` and `LightPalette` resources

conradolandia committed:
Add some minor corrections

conradolandia committed:
Add some caveats about inheriting from DarkPalette and LightPalette

conradolandia commented:
Thanks for your comments @dalthviz, this is again ready for review!

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -0,0 +1,459 @@
+-----------------
+
+1. Create Your Palette File
+~~~~~~~~~~~~~~~~~~~~~~~~~~
```
A warning related with the underline being to short here:

```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

dalthviz commented on qdarkstyle/__init__.py:
```diff
@@ -1,12 +1,22 @@
+**C++ usage:** Use the generated .qss files directly in your Qt C++ applications.
+
+**Note:** Python 2 and Qt4 (PyQt4, PySide) are no longer supported.
+For legacy support, use QDarkStyle version 3.1 or earlier.
```
I think this change also should be done to the `README.rst`. Probably a general check to the `README.rst` content could be worthy as part of this PR (like adding a custom palette section)

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -0,0 +1,459 @@
+
+
+Color Selection Guidelines
+~~~~~~~~~~~~~~~~~~~~~~~~~
```
A warning related with the underline being to short here:

```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~
```

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -0,0 +1,459 @@
+5. **System Colors**: Consider using built-in color constants for consistency
+
+Theme Development Workflow
+~~~~~~~~~~~~~~~~~~~~~~~~~
```
A warning related with the underline being to short here:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~
```

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -0,0 +1,459 @@
+5. **Validate**: Check accessibility and usability
+
+File Management
+~~~~~~~~~~~~~~
```
A warning related with the underline being to short here:
```suggestion
~~~~~~~~~~~~~~~
```

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -0,0 +1,459 @@
+4. **Backup**: Maintain copies of working themes
+
+Important Notes
+--------------
```
A warning related with the underline being to short here:
```suggestion
---------------
```

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -0,0 +1,459 @@
+--------------
+
+Limitations and Considerations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
A warning related with the underline being to short here:
```suggestion
Limitations and Considerations
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -0,0 +1,459 @@
+**Why load_stylesheet() doesn't support custom palettes**: The function validates palette IDs and only accepts 'dark' or 'light'. This is because the function needs to import the correct resource modules (darkstyle_rc or lightstyle_rc) that contain the compiled QSS and icon resources. Custom palettes don't have these pre-compiled resources available in the package, so they must be generated separ ...
+
+Troubleshooting
+--------------
```
A warning related with the underline being to short here:
```suggestion
---------------
```

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -0,0 +1,459 @@
+--------------
+
+Common Issues
+~~~~~~~~~~~~
```
A warning related with the underline being to short here:
```suggestion
~~~~~~~~~~~~~
```

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -0,0 +1,459 @@
+    sys.path.insert(0, './my_themes')
+    
+    # Import the generated resources
+    import neon.neonstyle_rc  # This loads the resources
```
I think this manual alteration of `sys.path` shouldn't be suggested. Instead it should be mentioned that the custom theme assets should be available as a module inside the same current working directory from where they are being imported or make they should made available as a proper package in case the idea is to use them as part of multiple applications

dalthviz commented on docs/scripts/qdarkstyle_utils.rst:
```diff
@@ -0,0 +1,55 @@
+
+    python -m qdarkstyle.utils [OPTIONS]
+
+The tool converts ``.qrc`` files to ``_rc.py`` and ``.rcc`` files, generates images from SVG sources, and compiles SCSS to QSS stylesheets.
```
I would say that a more accurate description would be something like:

```suggestion
The tool generates ``.qrc`` files,  ``_rc.py`` and ``.rcc`` files, generates images from SVG sources, and compiles SCSS to QSS stylesheets.
```

dalthviz commented on docs/scripts/qdarkstyle_utils.rst:
```diff
@@ -0,0 +1,55 @@
+QDarkStyle Utils CLI
+====================
+
+Command-line interface for processing QRC files, generating stylesheets, and creating custom palettes.
```
I think this should be something like
```suggestion
Command-line interface for processing palette class definitions, generating stylesheets and other palette assets, with support for custom palettes.
```

dalthviz reviewed (commented):
Thanks for the updates @conradolandia ! Left some further comments/suggestions

conradolandia committed:
Fix some formatting issues, improve README.rst and improve requirements descriptions. Add explicit documentation about the creation of Dark and Light variations for custom themes

conradolandia committed:
Apply suggestions from code review

Co-authored-by: Daniel Althviz Moré <16781833+dalthviz@users.noreply.github.com>

conradolandia committed:
Improved palette generation documentation regarding DarkPalette and LightPalette

conradolandia commented:
Thank you for your comments, @dalthviz, this is ready again if you want to check it out

conradolandia committed:
Solve some documentation issues regarding palette IDs

conradolandia committed:
Improve usage example in documentation

conradolandia committed:
Add demo for custom theme creation and update CONTRIBUTING.rst

conradolandia commented:
Thanks @dalthviz, I updated the documentation taking into account your comments, updated CONTRIBUTING.rst and also added a new file with a demo for custom theme creation.

conradolandia committed:
Add more documentation about the custom themes demo

dalthviz commented on qdarkstyle/__init__.py:
```diff
@@ -278,6 +283,8 @@ def _load_stylesheet(qt_api='', palette=None):
         palette = LightPalette
     else:
         print("Not recognized ID for palette! Exiting!")
+        print("Only 'dark' and 'light' palette IDs are supported by load_stylesheet().")
```
I would say the limitation here is more like:

```suggestion
        print("Only 'DarkPalette' and 'LightPalette' palettes (with 'dark' and 'light' IDs respectively) are supported by load_stylesheet().")
```

dalthviz commented on docs/custom_palettes.rst:
```diff
@@ -410,24 +626,28 @@ File Management
 
 **Why load_stylesheet() doesn't support custom palettes**: The function validates palette IDs and only accepts 'dark' or 'light'. This is because the function needs to import the correct resource modules (darkstyle_rc or lightstyle_rc) that contain the compiled QSS and icon resources. Custom palettes don't have these pre-compiled resources available in the package, so they must be generated separ ...
 
+**Future Compatibility**: By using standard 'dark' and 'light' IDs, custom themes follow the same architecture as built-in themes. This design choice makes it easier to potentially integrate custom themes with the main API in future versions while maintaining backward compatibility.
```
This probably will depend on plans for the future of the package, maybe you discussed this with @ccordoba12 ?

conradolandia replied:
We talked about that before, I am of the position that a theme should provide both variations.

conradolandia committed:
Update qdarkstyle/__init__.py

Co-authored-by: Daniel Althviz Moré <16781833+dalthviz@users.noreply.github.com>
